test/GridTest.cpp

```diff
@@ -28,11 +28,22 @@ namespace rtt{
         Grid grid = Grid(2, 2, 1, 1, 4, 4);
         for (auto nestedPoints : grid.getPoints()) {
             for (auto point : nestedPoints) {
-                EXPECT_TRUE(point.x >= 2 && point.x < 3);
-                EXPECT_TRUE(point.y >= 2 && point.y < 3);
+                EXPECT_TRUE(point.x >= 2 && point.x <= 3);
+                EXPECT_TRUE(point.y >= 2 && point.y <= 3);
             }
         }
     }
+    TEST(Grid,getters){
+        Grid grid = Grid(1,2,3,4,5,6);
+        EXPECT_EQ(grid.getOffSetX(),1);
+        EXPECT_EQ(grid.getOffSetY(),2);
+        EXPECT_EQ(grid.getRegionWidth(),3);
+        EXPECT_EQ(grid.getRegionHeight(),4);
+        EXPECT_EQ(grid.getNumSegmentsX(),5);
+        EXPECT_EQ(grid.getNumSegmentsY(),6);
+        EXPECT_DOUBLE_EQ(grid.getStepSizeX(),3/5.0); //TOOD: possible bugs? Is the grid defined to be inclusive edge or not?
+        EXPECT_DOUBLE_EQ(grid.getStepSizeY(),4/6.0);
```

Comment:
I think the grid should include both edges, so a grid from 0 to 1 divided into 4 segments would look like:
(0,1) (0.25,1) (0.5,1) (0.75, 1) (1,1)
(0, 0.75) (0.25, 0.75) (0.5,0.75) (0.75, 0.75) (1,0.75)
(0, 0.5) (0.25, 0.5) (0.5, 0.5) (0.75, 0.5) (1, 0.5)
(0, 0.25) (0.25, 0.25) (0.5, 0.25) (0.75, 0.25) (1, 0.25)
(0, 0) (0.25, 0) (0.5, 0) (0.75, 0) (1, 0)

That's what I had in mind when making this class (and unless I miscalculated is also how it functions in this PR) but of course it's personal preference

Reply:
Yes, that's exactly what it does now.